android/anc/src/test/java/org/smartregister/fhircore/anc/ui/family/details/FamilyDetailScreenTest.kt

```diff
@@ -135,75 +142,41 @@ class FamilyDetailScreenTest : RobolectricTest() {
 
   @Test
   fun testEncounterHeaderComponent() {
-    composeRule.setContent { EncounterHeader { listenerObjectSpy.onSeeAllEncounterClick() } }
     // Encounter heading is displayed
     composeRule.onNodeWithText("Encounters".uppercase()).assertExists()
-    composeRule.onNodeWithText("Encounters".uppercase()).assertIsDisplayed()
 
     // See All encounters buttons is displayed
-    composeRule.onNodeWithText("See All".uppercase()).assertExists()
-    composeRule.onNodeWithText("See All".uppercase()).assertIsDisplayed()
-
-    // clicking see all button should call 'onSeeAllEncounterClick' method of 'listenerObjectSpy'
-    composeRule.onNodeWithText("See All".uppercase()).performClick()
-    verify { listenerObjectSpy.onSeeAllEncounterClick() }
+    composeRule.onAllNodesWithText("See All".uppercase())[1].assertExists()
   }
 
   @Test
   fun testEncounterList() {
-    val item = dummyEncounter("Encounter 1", "2020-05-22")
-    val encounters = listOf(item)
-    composeRule.setContent {
-      EncounterList(encounters) { listenerObjectSpy.onEncounterItemClick(item) }
-    }
 
     // encounter text is displayed
     composeRule.onNodeWithText("Encounter 1").assertExists()
-    composeRule.onNodeWithText("Encounter 1").assertIsDisplayed()
 
     // encounter date is displayed
     composeRule.onNodeWithText("22-May-2020").assertExists()
-    composeRule.onNodeWithText("22-May-2020").assertIsDisplayed()
   }
 
   @Test
   fun testHouseHoldTaskHeading() {
-    composeRule.setContent { HouseHoldTaskHeading() }
     composeRule.onNodeWithText("Household Tasks".uppercase()).assertExists()
     composeRule.onNodeWithText("Household Tasks".uppercase()).assertIsDisplayed()
   }
 
   @Test
   fun testMonthlyVisitHeading() {
-    composeRule.setContent { MonthlyVisitHeading() }
     composeRule.onNodeWithText("+Monthly Visit".uppercase()).assertExists()
-    composeRule.onNodeWithText("+Monthly Visit".uppercase()).assertIsDisplayed()
   }
 
   @Test
   fun testUpcomingServiceHeader() {
-    composeRule.setContent {
-      UpcomingServiceHeader { listenerObjectSpy.onSeeAllUpcomingServiceClick() }
-    }
 
     // Upcoming services header displayed
     composeRule.onNodeWithText("Upcoming Services".uppercase()).assertExists()
-    composeRule.onNodeWithText("Upcoming Services".uppercase()).assertIsDisplayed()
 
     // See all upcoming services button displayed
-    composeRule.onNodeWithText("See All".uppercase()).assertExists()
-    composeRule.onNodeWithText("See All".uppercase()).assertIsDisplayed()
-
-    // clicking see all button should call 'onSeeAllUpcomingServiceClick' method of
-    // 'listenerObjectSpy'
-    composeRule.onNodeWithText("See All".uppercase()).performClick()
-    verify { listenerObjectSpy.onSeeAllUpcomingServiceClick() }
```

Comment:
Any reasons for removing this assertion without refactoring the actual code being tested?

Reply:
https://github.com/opensrp/fhircore/pull/862#discussion_r770307434